### PR TITLE
cmd/tailscale/cli: add debug set-expire command for testing

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#8811
Updates tailscale/corp#8613

### CODE REVIEW

File: cmd/tailscale/cli/debug.go [hunk 1]
```diff
@@ -166,6 +166,16 @@ var debugCmd = &ffcli.Command{
 				return fs
 			})(),
 		},
+		{
+			Name:      "set-expire",
+			Exec:      runSetExpire,
+			ShortHelp: "manipulate node key expiry for testing",
+			FlagSet: (func() *flag.FlagSet {
+				fs := newFlagSet("set-expire")
+				fs.DurationVar(&setExpireArgs.in, "in", 0, "if non-zero, set node key to expire this duration from now")
```
Comment (nickoneill) [inline cmd/tailscale/cli/debug.go:175]: Can we indicate that duration is seconds and not ms/h/d?

Comment (bradfitz) [reply cmd/tailscale/cli/debug.go:175]: those all work. It's a duration flag var, so it requires a unit suffix.

Comment (nickoneill) [reply cmd/tailscale/cli/debug.go:175]: 👍

File: client/tailscale/localclient.go [hunk 2]
```diff
@@ -1020,6 +1020,15 @@ func (lc *LocalClient) DebugDERPRegion(ctx context.Context, regionIDOrCode strin
 	return decodeJSON[*ipnstate.DebugDERPRegionReport](body)
 }
 
+// DebugSetExpireIn marks the current node key to expire in d.
+//
+// This is meant primarily for debug and testing.
+func (lc *LocalClient) DebugSetExpireIn(ctx context.Context, d time.Duration) error {
+	v := url.Values{"expiry": {fmt.Sprint(time.Now().Add(d).Unix())}}
+	_, err := lc.send(ctx, "POST", "/localapi/v0/set-expiry-sooner?"+v.Encode(), 200, nil)
```
Comment (maisem) [inline client/tailscale/localclient.go:1028]: I don't think this endpoint does any auth, can you pls fix?

Comment (bradfitz) [reply client/tailscale/localclient.go:1028]: Discussed on chat. Also that endpoint isn't new.

Comment (bradfitz) [reply client/tailscale/localclient.go:1028]: Done, though. serveLogout required Write access as-is, so now they match.

**Review by maisem — APPROVED** [review]

**Pull request merged**